pkg/odo/cli/dev/dev.go

```diff
@@ -105,6 +106,18 @@ func (o *DevOptions) PreInit() string {
 }
 
 func (o *DevOptions) Complete(ctx context.Context, cmdline cmdline.Cmdline, args []string) error {
+	if !o.noCommandsFlag {
+		devfileObj := *odocontext.GetEffectiveDevfileObj(ctx)
+		hasCommands, err := libdevfile.HasCommands(devfileObj.Data)
```

Comment:
I think it would be more robust and adaptable if the existence of commands are checked separately when we try to run them. For example, if no Build command exists but a Run command exists, of the other way.

There is also the case when there is not default command, but the user passes the Build or Run command to consider using `--build-command` and/or `--run-command`